File: CriminalIntent/app/src/main/java/com/bignerdranch/android/criminalintent/database/CrimeDao.java
package com.bignerdranch.android.criminalintent.database;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Query;

import com.bignerdranch.android.criminalintent.Crime;

import java.util.List;
import java.util.UUID;

@Dao
public interface CrimeDao {

    @Query("select * from Crime")
    LiveData<List<Crime>> getCrimes();
    @Query("select * from Crime where id = :id")
    LiveData<Crime> getCrime(UUID id);
}

File: CriminalIntent/app/src/main/java/com/bignerdranch/android/criminalintent/database/CrimeDao.kt
package com.bignerdranch.android.criminalintent.database

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Query
import com.bignerdranch.android.criminalintent.Crime
import java.util.*

@Dao
interface CrimeDao {

    @Query("select * from Crime")
    fun getCrimes(): LiveData<List<Crime>>;
    @Query("select * from crime where id = (:id)")
    fun getCrime(id: UUID): LiveData<Crime?>;
}
